src/wallet/asyncrpcoperation_sendmany.cpp

```diff
@@ -354,6 +365,141 @@ bool AsyncRPCOperation_sendmany::main_impl() {
     LogPrint("zrpcunsafe", "%s: private output: %s\n", getId(), FormatMoney(z_outputs_total));
     LogPrint("zrpc", "%s: fee: %s\n", getId(), FormatMoney(minersFee));
 
+
+    /**
+     * SCENARIO #0
+     *
+     * Sprout not involved, so we just use the TransactionBuilder and we're done.
+     * We added the transparent inputs to the builder earlier.
+     */
+    if (isUsingBuilder_) {
+        builder_.SetFee(minersFee);
+
+        // Get various necessary keys
+        SaplingExpandedSpendingKey expsk;
+        SaplingFullViewingKey from;
+        if (isfromzaddr_) {
+            auto sk = boost::get<libzcash::SaplingSpendingKey>(spendingkey_);
+            expsk = sk.expanded_spending_key();
+            from = expsk.full_viewing_key();
+        } else {
+            // TODO: Set "from" to something!
```

Comment:
Should this be done as part of this PR?

Reply:
It needs to be done as part of 2.0.1, as otherwise it is just using the empty FVK's ovk, which IIRC is the all-zeroes byte string. @daira suggested that we use the FVK of one of the recipients, but that only works if one of the recipients is us, which is not guaranteed. I'm wondering whether we should define a way to derive a synthetic ovk from a transparent spending key, so that the semantics (`from` corresponds to sender's address) are maintained?

Reply:
Filed https://github.com/zcash/zcash/issues/3506

Reply:
Wait when did I suggest that? I don't remember doing so. I recommend using the FVK of the ZIP 32 account that sent the transfer.